### PR TITLE
integrations: DD_SERVICE

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This PR makes the following changes:

1. When `DD_SERVICE`/`config.service` is provided it is the default service name used by non-external-calling integrations. The expected behaviour when not using `DD_SERVICE`/`config.service` should remain as before.

2. Integrations that never used `{DD,DATADOG}_SERVICE_NAME` should use `config.service` instead of `config.get_service` since using the latter would introduce new functionality for integrations to use the old `{DD,DATADOG}_SERVICE_NAME`.

3. `config.get_service` refactored to `config._get_service`. This should be a private api used only in integrations which use `{DD,DATADOG}_SERVICE_NAME` in order to deprecate the usage.

### CODE REVIEW

File: ddtrace/contrib/requests/patch.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 # requests default settings
 config._add('requests', {
-    'service_name': get_env('requests', 'service_name', default=DEFAULT_SERVICE),
+    "service_name": config.get_service(default=get_env("requests", "service_name", default=DEFAULT_SERVICE)),
```
Comment (jdgumz) [inline ddtrace/contrib/requests/patch.py]: Is this _just_ for server-side? Or will this apply to all clients that use `requests`?

File: tests/contrib/requests/test_requests.py [hunk 2]
```diff
@@ -492,3 +492,14 @@ def test_analytics_integration_on_using_pin_default(self):
         self.assertEqual(len(spans), 1)
         s = spans[0]
         self.assertEqual(s.get_metric(ANALYTICS_SAMPLE_RATE_KEY), 1.0)
+
+    @BaseTracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
```
Comment (jdgumz) [inline tests/contrib/requests/test_requests.py]: I think we should have a test where we have both the `DD_SERVICE` and `DD_SERVICE_NAME` provided. Folks could definitely run into that case when they migrate.

I don't think this is relevant for all integrations, just the ones where you have
```
config.get_service(default=get_env(...))
```

Comment (Kyle-Verhoog) [reply tests/contrib/requests/test_requests.py]: `config.get_service(...)` deals with also checking `DD_SERVICE_NAME` (and `DATADOG_SERVICE_NAME`). Although I'm not sure if I added a test for `config.get_service()` to ensure this. So I'll make sure to do that.

Do you think that's sufficient or should we still have tests on each integration to ensure the compatibility?

File: ddtrace/contrib/requests/patch.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/requests/patch.py]: This will apply to `requests` in general. Let me know if this is not what we want 😛

File: ddtrace/settings/config.py [hunk 3]
```diff
@@ -32,7 +32,11 @@ def __init__(self):
         )
 
         self.env = get_env("env")
+        # get_env will get DD_SERVICE or DATADOG_SERVICE (deprecated)
+        # Note that we do not call self.get_service() here because we never
+        # supported {DD,DATADOG}_SERVICE_NAME for setting the service globally.
```
Comment (Kyle-Verhoog) [inline ddtrace/settings/config.py]: @brettlangdon @jdgumz should we support using `{DD,DATADOG}_SERVICE_NAME` exactly the same as the new `DD_SERVICE`? Currently, I opted to maintain compatibility so there would be no change for existing configurations using `{DD,DATADOG}_SERVICE_NAME`.

This could be confusing either way. Thoughts?

Comment (jdgumz) [reply ddtrace/settings/config.py]: Maybe we could add a precedence order? I think this might be baked into `get_env` already, but basically between `DD_` and `DATADOG_` we'd prefer the former over the latter.

That way it doesn't break anything existing, it only "breaks" when you opt into using the new variable format.

File: tests/contrib/requests/test_requests.py (hunk 2 above)
Comment (jdgumz) [reply tests/contrib/requests/test_requests.py]: Yeah it might be covered by calling `config.get_service()` with one/both of the env vars provided.

File: ddtrace/contrib/requests/patch.py (hunk 1 above)
Comment (jdgumz) [reply ddtrace/contrib/requests/patch.py]: cc @brettlangdon I'm pretty sure we don't want the requests client's service name to be overridden (not yet anyway).

Also, I've never used `requests` as a server-side framework before. I didn't know you could?

File: ddtrace/settings/config.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply ddtrace/settings/config.py]: Yeah the precedence exists in that order already 😄 

The "breaking" I'm thinking of is that we used to only use `{DD,DATADOG}_SERVICE_NAME` for web-framework integrations. If we treat `{DD,DATADOG}_SERVICE_NAME` the same as `{DD,DATADOG}_SERVICE` then existing users will see a whole bunch more collapsed under that service name.

If we keep `.._SERVICE_NAME` and `.._SERVICE` the same then we're only introducing new functionality, not changing anything existing.

Comment (jdgumz) [reply ddtrace/settings/config.py]: Ah yeah I see then. We want to keep them separate. `_SERVICE_NAME` should be considered deprecated and will not get the same functionality as `_SERVICE`.

File: ddtrace/settings/config.py [hunk 4]
```diff
@@ -32,7 +32,11 @@ def __init__(self):
         )
 
         self.env = get_env("env")
+        # get_env will get DD_SERVICE or DATADOG_SERVICE (deprecated)
+        # Note that we do not call self.get_service() here because we never
```
Comment (jdgumz) [inline ddtrace/settings/config.py]: nit: maybe reword to `because we do not support ...`?

File: tests/contrib/kombu/test.py [hunk 5]
```diff
@@ -117,3 +117,49 @@ def test_analytics_without_rate(self):
...
+                              declare=[task_queue])
+
+        with kombu.Consumer(self.conn, [task_queue], accept=['json'], callbacks=[process_message]) as consumer:
+            Pin.override(consumer, tracer=self.tracer)
+            self.conn.drain_events(timeout=2)
+
+        spans = self.get_spans()
+        self.assertEqual(len(spans), 2)
+        for span in spans:
+            assert span.service == "mysvc"
```
Comment (jdgumz) [inline tests/contrib/kombu/test.py]: Do these spans include the producer span?

I think we only want the consumer span to have `DD_SERVICE` inherited.

Although maybe it makes sense to have both? @brettlangdon what do you think?

Comment (Kyle-Verhoog) [reply tests/contrib/kombu/test.py]: Just sync'd with @brettlangdon who agrees that we should treat the producer as an external service call.

File: ddtrace/contrib/kombu/patch.py [hunk 6]
```diff
@@ -45,9 +46,13 @@ def patch():
...
     _w('kombu', 'Producer._publish', traced_publish)
     _w('kombu', 'Consumer.receive', traced_receive)
+
+    # We do not provide a service for producer spans since they represent
+    # external service calls to another service.
+    # Instead the service should be inherited from the parent.
     Pin(
-        service=config.kombu['service_name'],
-        app='kombu'
+        service=None,
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/kombu/patch.py]: Note that this is a breaking change for kombu, since consumer spans will now inherit the service from the parent instead of being defined by DD_KOMBU_SERVICE_NAME or `DEFAULT_SERVICE`.

@brettlangdon are we okay with this?

We _could_ do something like this for backwards compatibility... but then consumer spans will _never_ inherit from the parent since the default `DEFAULT_SERVICE` is provided 😕.  

```suggestion
        service=get_env("kombu", "service_name", default=DEFAULT_SERVICE), 
```

Comment (brettlangdon) [reply ddtrace/contrib/kombu/patch.py]: I think if we do it where it only inherits **if** `DD_SERVICE`/global service is defined, then the "breaking change" is ok, since it is opt in.
